src/scripts/main.js

```diff
@@ -1 +1,73 @@
 'use strict';
+
+const firstPromise = new Promise((resolve, reject) => {
+  const handler = () => {
+    resolve();
+    document.addEventListener('click', resolve);
```

Comment:
Here, you are adding an event listener inside the handler for the click event. This means that every time a click event occurs, a new event listener is added. This could lead to a memory leak. Consider removing the event listener once it has been triggered.